packages/framework/fluid-static/api-report/fluid-static.api.md

```diff
@@ -15,6 +15,9 @@ import { IFluidLoadable } from '@fluidframework/core-interfaces';
 import { IRuntimeFactory } from '@fluidframework/container-definitions/internal';
 import { ISharedObjectKind } from '@fluidframework/shared-object-base';
 
+// @public
+export type CompatMode = "1" | "2";
```

Comment:
I'd make it an object / interface, not a string.
We will need to expose various options (like compression knobs) as our customers (like Autodesk) will ask for it.

There is no better place to do it other than here, on an object that is unique to a compat mode.
In other words, it should not be possible (should be compile error) for users to attempt to set or change compression settings for "1" schema as it does not apply / does not exist at that level of compatibility.

Same in reverse - when/if we remove ability some of these knobs in the future (let's say in 3.0 compat mode), ideally it should be a compile break for users do change these knobs on config that no longer allows that.

I also think it's more readable if it says { compatLevel: "1" } over "1" (even though more verbose).

Reply:
My intention is to disallow nuanced configuration here - in the sync we agreed this feature is purely about migrating from 1->2, not for configuring other accessory features.  Those features should design and establish their own appropriate API surface to enable/disable/customize.

E.g. GC uses the MonitoringContext to enable, or summaryCompression is offered on AzureClientProps.  There might be better options available for how to do it and we can certainly discuss, but it's a separate scope from the primary concern of safely getting customers from 1->2.

Reply:
I think these are not great examples (there is no overlap).
Let's play the game in reverse - let's assume there is already compression setting out there and you are adding this PR. What would you do if user specifies compression and V1? Would it throw UsageError()? Let it use compression despite user saying - use compatibility mode?

Reply:
Given significantly complicated compat story and upgrade paths, I generally do like the idea to simplify and constrain in this case. Simplify: if the features are tightly coupled, offer them through unified APIs. Constrain: User should have very limited and explicit means to toggle/config anything related to our incompatible features.